hpx/lcos/promise.hpp

```diff
@@ -167,97 +167,97 @@ namespace lcos {
             > base_type;
 
     public:
-        // Effects: constructs a promise object and a shared state.
+        /// \brief constructs a promise object and a shared state.
         promise()
           : base_type()
         {}
 
-        // Effects: constructs a promise object and a shared state. The
-        // constructor uses the allocator a to allocate the memory for the
-        // shared state.
+        /// \brief constructs a promise object and a shared state. The
+        /// constructor uses the allocator a to allocate the memory for the
+        /// shared state.
         template <typename Allocator>
         promise(std::allocator_arg_t, Allocator const& a)
           : base_type(std::allocator_arg, a)
         {}
 
-        // Effects: constructs a new promise object and transfers ownership of
-        //          the shared state of other (if any) to the newly-
-        //          constructed object.
-        // Postcondition: other has no shared state.
+        /// \brief constructs a new promise object and transfers ownership of
+        ///          the shared state of other (if any) to the newly-
+        ///          constructed object.
+        /// Postcondition: other has no shared state.
         promise(promise&& other) noexcept : base_type(std::move(other))
         {}
 
-        // Effects: Abandons any shared state
+        /// \brief Abandons any shared state
         ~promise()
         {}
 
-        // Effects: Abandons any shared state (30.6.4) and then as if
-        //          promise(std::move(other)).swap(*this).
-        // Returns: *this.
+        /// \brief Abandons any shared state (30.6.4) and then as if
+        ///          promise(std::move(other)).swap(*this).
+        /// \return *this.
```

Comment:
Please be consistent with `\return` and `\returns`. I think I prefer `\returns` as it reads more naturally (and it's what we use in most places in the codebase).